File: app/models/taxon.rb
# == Schema Information
#
# Table name: taxons
#
#  id          :integer          not null, primary key
#  name        :string(120)      not null
#  rank        :string(16)
#  parent_id   :integer
#  valid_taxon :binary(1)        default("b'1'"), not null
#

class Taxon < ApplicationRecord
  include ReadOnlyModel

  belongs_to :lineage, foreign_key: 'id', primary_key: 'taxon_id', class_name: 'Lineage'

  scope :with_genome, -> { distinct.joins('RIGHT JOIN uniprot_entries ON taxons.id = uniprot_entries.taxon_id') }

  # sorting order
  def <=>(other)
    return -1 if other.nil?

    id <=> other.id
  end
end

File: backend/java/src/main/java/org/unipept/storage/Taxon.java
package org.unipept.storage;

public class Taxon {

    public String rank;
    public String name;
    public int parentId;
    public boolean validTaxon;
    public int lineageCount;
    public boolean used;

    public Taxon(int id, String name, String rank, int parentId, boolean validTaxon) {
        this.name = name;
        this.rank = rank;
        this.parentId = parentId;
        this.lineageCount = 0;
        this.validTaxon = validTaxon;
        this.used = false;
    }

}
